src/components/templates/search-modal/index.tsx

```diff
@@ -0,0 +1,167 @@
+import {
+  useAdminCustomers,
+  useAdminDiscounts,
+  useAdminOrders,
+  useAdminProducts,
+} from "medusa-react"
+import React from "react"
+import { useDebounce } from "../../../hooks/use-debounce"
+import Spinner from "../../atoms/spinner"
+import Input from "../../atoms/text-input"
+import SearchIcon from "../../fundamentals/icons/search-icon"
+import * as RadixDialog from "@radix-ui/react-dialog"
+import CustomerResults from "./results/customer-results"
+import DiscountResults from "./results/discount-results"
+import KeyboardShortcuts from "./keyboard-shortcuts"
+import ProductResults from "./results/product-results"
+import useKeyboardNavigationList from "./use-keyboard-navigation-list"
+import clsx from "clsx"
+import OrderResults from "./results/order-results"
+import CrossIcon from "../../fundamentals/icons/cross-icon"
+import Tooltip from "../../atoms/tooltip"
+
+const getTotal = (...lists) =>
+  lists.reduce((total, list = []) => total + list.length, 0)
+
+const SearchModal = ({ handleClose }) => {
+  const [q, setQ] = React.useState("")
+  const query = useDebounce(q, 500)
+  const onChange = (e) => setQ(e.target.value)
+  const handleClear = () => {
+    setQ("")
+  }
+
+  const { orders, isFetching: isFetchingOrders } = useAdminOrders(
+    {
+      display_id: query,
+      limit: 5,
+      offset: 0,
+    },
+    { enabled: !!query, keepPreviousData: true }
+  )
```

Comment:
Don't know if I have confused things a bit but let me try to explain two stories that I think we should solve for when querying for orders:

## Story 1
- I am a customer service user and I get an email from a customer who sends me her order id (i.e. `order.display_id`) - I copy the order id 
- If I do `cmd + k` then `cmd + v` that should take me straight to the order - I don't necessarily want to wait for all the other queries to complete.

## Story 2
- I am receiving a call from a customer and they don't have their order number but they give me their email
- I want to do `cmd + k` and start typing their email; I want to see their most recent order or alternatively just their customer profile

Right now the order query here doesn't solve for Story 2; my thoughts on how we could accomplish both stories would be this:

Get orders by the free text search i.e.:
```suggestion
  const { orders, isFetching: isFetchingOrders } = useAdminOrders(
    {
      q: query,
      limit: 5,
      offset: 0,
    },
    { enabled: !!query, keepPreviousData: true }
  )
```

Additionally add another query like this one:
```js
  const { orders: directHitOrders, isFetching: isFetchingOrders } = useAdminOrders(
    {
      display_id: query,
      limit: 5,
      offset: 0,
    },
    { enabled: !!query, keepPreviousData: true }
  )
```

If we have a hit on i.e. `directHitOrders.length === 1 && directHitOrders[0].display_id === query` then take us straight to the order else ignore `directHitOrders`

--- 

The stuff above was what my initial thought was what I am realizing now, however, is that if we just go with the suggestion above my "cursor" will already be on the order I want to go to so all I would have to do would be to hit `enter` and I would get there - I think that is a fair flow to stick with so no need to add the additional query. 

**So in conclusion** let's allow users to also search orders by email i.e. change `display_id` to `q`. I realize this is a lot of talk for a simple fix but just wanted to open-source my thinking :D

Reply:
thanks for taking the time to explain @srindom! I also think the flow we have rn should be good enough, although what you mentioned in Story 1 could be super useful!

Although one edge case I think we should be a tiny bit careful about is one where the order number conflicts with something like a discount code. For example, user types in 10 initially to look for a `DISCOUNT10`, if it turns out that there is an order with display_id 10, then we'd always be prioritizing the order, even though the user is looking for a discount. This is probably a bit of a stretch because the probability might be very small, but could still happen I guess :sweat_smile: